splinterd/src/main.rs

```diff
@@ -617,6 +619,42 @@ pub enum GetTransportError {
     IoError(io::Error),
 }
 
+impl Error for GetTransportError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            GetTransportError::CertError(_) => None,
+            GetTransportError::NotSupportedError(_) => None,
+            GetTransportError::TlsTransportError(err) => Some(err),
+            #[cfg(feature = "generate-certs")]
+            GetTransportError::OpensslError(err) => Some(err),
+            GetTransportError::IoError(err) => Some(err),
+        }
+    }
+}
+
+impl fmt::Display for GetTransportError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            GetTransportError::CertError(msg) => {
+                write!(f, "ran into an error retrieving certificate: {}", msg)
+            }
+            GetTransportError::NotSupportedError(msg) => {
+                write!(f, "recieved transport type that is not supported: {}", msg)
+            }
+            GetTransportError::TlsTransportError(err) => {
+                write!(f, "unable to create TLS transport: {:?}", err)
+            }
+            #[cfg(feature = "generate-certs")]
+            GetTransportError::OpensslError(err) => {
+                write!(f, "unable to generate certificatesc: {}", err)
```

Comment:
Looks like a typo ("certificatesc"). Also, is cert generation the only reason for an `OpensslError` could occur?

Reply:
Its the only place that uses openssl in the splinter daemon code.